src/ucp/api/ucp.h

```diff
@@ -560,15 +560,17 @@ typedef enum {
      * Indicates that the data provided in @ref ucp_am_data_callback_t callback
      * can be held by the user. If UCS_INPROGRESS is returned from the callback,
      * the data parameter will persist and the user has to call
-     * @ref ucp_am_data_release when data is no longer needed.
+     * @ref ucp_am_data_release when data is no longer needed. This flag is
+     * mutually exclusive with @a UCP_AM_RECV_ATTR_FLAG_RNDV.
      */
-    UCP_AM_RECV_ATTR_FLAG_DATA         = UCS_BIT(16),
+    UCP_AM_RECV_ATTR_FLAG_DEFER_DATA   = UCS_BIT(16),
```

Comment:
few more naming options:
UCP_AM_RECV_ATTR_FLAG_DEFERABLE_DATA
UCP_AM_RECV_ATTR_FLAG_CAN_KEEP_DATA
UCP_AM_RECV_ATTR_FLAG_CAN_DETACH_DATA
UCP_AM_RECV_ATTR_FLAG_CAN_HOLD_DATA

Reply:
If we are adding API similar to UCT, why just don't reuse naming for similar functionality as well?
https://github.com/openucx/ucx/blob/bbf159e439e9c45678e6ceb6252d45811aead8fd/src/uct/api/uct_def.h#L67

Reply:
> If we are adding API similar to UCT, why just don't reuse naming for similar functionality as well?
> https://github.com/openucx/ucx/blob/bbf159e439e9c45678e6ceb6252d45811aead8fd/src/uct/api/uct_def.h#L67

yes, this is also an option